enhancements/last-scanned-timestamps.md

```diff
@@ -0,0 +1,87 @@
+---
+title: last-scanned-timestamps
+authors:
+  - TBD
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - @rhmdnd
+approvers:
+  - TBD
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - TBD
+creation-date: 2024-04-10
+last-updated: 2024-04-10
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - TBD
+see-also:
+  - /
+replaces:
+  - /
+superseded-by:
+  - /
+---
+
+# Add `last_scanned_timestamp` to `compliancecheckresults.compliance.openshift.io` objects
+
+## Summary
+
+ACS implements an integration with Compliance Operator. For this, ACS reports when a scan was completed. Currently
```

Comment:
I think the part that makes this tricky is that the data is only flowing one way (from sensor to central). Without a definitive signal that tells central "yep, you have everything" it's required to make an assumption (e.g., I've waited 5 minutes without anything new coming in from this particular suite, which is marked as DONE, I should be safe to build a report?).

Reply:
We could also implement like this:
After a scan report is triggered, wait until ComplianceSuite is in state `DONE`, than read all results and check that the `lastScannedTimestamp` is after the last triggered scan time. 
If not, retry 10 minutes later.
Would require a small go routine background job though to manage this.

@rhmdnd How likely do you even think it is that a `result` is send **after** the suite reports `DONE`? 
Imho we could also implement a small buffer of some minutes after which the the report is sent.

Reply:
Not just a report problem.  ACS simply has no way to know if the results match an iteration of a scan.

Reply:
> Not just a report problem. ACS simply has no way to know if the results match an iteration of a scan.

Is a guarantee necessary or the current implementation "good enough"?

If a guarantee is necessary we would need some kind of a UUID per CO scan.

Reply:
What if the ComplianceScan object contained a counter with the number of ComplianceCheckResults it generated the most recent run? Then consumers would have a known quantity to check for.

Reply:
@rhmdnd Is the ComplianceScan object aware of the total check results it creates?

Reply:
As an integer, it isn't. We could implement that though in the Reconcile loop of the ComplianceScan and tack that number onto the scan.